### DISCUSSION TITLE
HM-MOD-RPI-PCB an den PC - möglich?

### DISCUSSION THREAD

**Discussion by cablesky** [OpenCCU/OpenCCU#2015] [🇩🇪 Fragen&Antworten] [answered]
Hallo!
Kennt jemand eine günstige Möglichkeit den HM-MOD-RPI-PCB an einen PC (i386) anzuschließen? 
Ok, es gibt ja HB-RF-USB und HB-RF-USB-2 - aber hier kosten allein die Platinen schon ca. 30 Euro... 
Ginge das auch z.B. über ein FTDI232 oder ähnliches? 

Ich suche nämlich gerade eine Alternative zum Raspberry Pi, und da gibt es aktuell günstige Thin-Clients...

**Comment by jens-maus** [comment, marked as answer]
> Ginge das auch z.B. über ein FTDI232 oder ähnliches? 

Das wird nix. Alle Homematic Funkmodule brauchen spezielle Kerneltreiber (generic_raw_uart). Mit einem generischen FTDI wird das nix werden! Wurde auch schon mehrfach diskutiert, besprochen, analysiert!

Ich würde einfach zu einem HmIP-RFUSB raten den du via USB anschließen kannst. Fertig.

**Reply by cablesky** [reply]
Danke Dir für die Antwort!

**Reply by jp112sdl** [reply]
@cablesky Ja natürlich geht das.
Wenn du das HM MOD RPI PCB schon hast und evtl. einen FTDI, musst du nix neu kaufen.
Ist nix anderes als das hier: https://github.com/stan23/myPCBs/tree/master/HB-MOD-UART-USB
Nehme ich immer für Tests, zusammen mit RM in einer Virtualbox Umgebung

**Reply by cablesky** [reply]
@jp112sdl : Das ist sehr interessant! Hast du den FTDI irgendwie speziell konfiguriert (mit [FT Prog](https://ftdichip.com/utilities/#ft_prog) z.B. ?

**Reply by jp112sdl** [reply]
@cablesky Ja, du musst das Template https://github.com/alexreinert/PCB/blob/master/HB-RF-USB/HB-RF-USB_FT232R_Template.xml mittels FTProg einspielen

**Reply by opl-git** [reply]
> @cablesky Ja natürlich geht das. Wenn du das HM MOD RPI PCB schon hast und evtl. einen FTDI, musst du nix neu kaufen. Ist nix anderes als das hier: https://github.com/stan23/myPCBs/tree/master/HB-MOD-UART-USB Nehme ich immer für Tests, zusammen mit RM in einer Virtualbox Umgebung

@jp112sdl @cablesky 
Hallo und sorry, dass ich die alte Diskussion ausgrabe. Ich habe aktuell genau dieses Problem. Habe ein HM-MOD-RPI-PCB und einen FTDI, welche ich gerne in Raspberrymatric nutzen möchte.
Habe es so verstanden, dass ich auf den FTDI mittels FT Prog das Template (HB-RF-USB_FT232R_Template.xml) installieren muss. Dann würde ich die 4 Kabel (GND, VCC, TX, R) zwischen HM-MOD-RPI-PCB und FTDI verbinden.
Würde es so funktionieren?

Vielen Dank schon mal!

**Reply by jp112sdl** [reply]
Ja, ich glaub so war das und so müsste das funktionieren

**Reply by opl-git** [reply]
@jp112sdl  Danke für die Rückmeldung. 
Hatte das Template mittels FT Prog eingespielt. War auch nach 1 Sekunde fertig. 
Habe dann den FTDI mit dem HM-MOD-RPI-PCB nach folgender Anleitung verbunden: 
![R](https://github.com/user-attachments/assets/7299e7b1-a3b5-4f97-8c18-47858ebf54d1)
Am FTDIUSB Stick habe ich RX und TX gekreuzt.

Leider sagt mir Raspberrymatic in meiner Proxmox Umgegebung, dass es bei Klick auf Anlernen "... keine nutzbaren Homematic Funkmodule gefunden...". Den USB hatte ich in Proxmox durchgereicht.

**Comment by cablesky** [comment]
@jp112sdl vielen Dank für den Tipp, den probiere ich sofort aus! 👍
@jens-maus hoffe, du kommst dir nicht übergangen vor..

Geile arbeit von euch beiden! Hab von jedem von euch was in Betrieb! 😉

**Reply by jens-maus** [reply]
> @jens-maus hoffe, du kommst dir nicht übergangen vor..

Nein, alles gut. Das man einen FT232 mittels des von @jp112sdl benannten templates zu einem HB-RF-USB machen kann ist natürlich vollkommen ok, denn dann nutzt der ja den generic raw uart treiber. Trotzdem sollte man ggf überlegen ob es nicht der sache wert wäre sich solch ein HB-RF-USB-2 käuflich zu erwerben weil einfacher, getestet und so dann auch ein OpenHardware produkt unterstützt wird!

**Reply by cablesky** [reply]
Ich habe noch ein paar Fragen dazu:
1) erkennt RaspberryMatic den USB-Stick selbständig, oder müssen noch irgendwelche Kernel Module geladen werden? (Ich lasse HomeAssistant in einer Proxmox Umgebung mit weitergereichtem USB-Port)
2) Braucht man die Aschlüsse Reset und PWREn unbedingt?

**Comment by cablesky** [comment]
![threema-20221028-140927158](https://user-images.githubusercontent.com/28872858/198619177-9a5ebfbc-1c38-41bf-a7c0-bdeb1804a72e.jpg)
Versuchs Aufbau

**Reply by cablesky** [reply]
Ha! Mein Test-Aufbau funktioniert! Also wird Reset und PWREn nicht unbedingt benötigt! Genial, ich freu mich! :-)

**Reply by jens-maus** [reply]
> Also wird Reset und PWREn nicht unbedingt benötigt!

Reset wäre nicht schlecht, denn sonst kann RaspberryMatic beim durchbooten bzw reboot nicht das funkmodul aus einem komischen Status zurückholen!
